spec/sessions/sign_up_spec.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+require_relative '../helpers/spec_helper'
+require_relative '../helpers/auth_helper'
+require_relative '../helpers/logs_helper'
+require_relative '../../app/controllers/sessions_controller'
+require_relative '../../app/controllers/records_controller'
+require 'yaml'
+
+data = YAML.load_file('data.yml')
+
+RSpec.describe '[Sessions API]' do
+  include AuthHelper
+  include LoggerHelper
+
+  context 'when GET /sign_up_form' do
+    context 'when authorised' do
+      before(:all) { log_in('TestUser', 'Test123456!') }
+
+      it_behaves_like 'authorised get request', '/homepage' do
+        let(:title) { data['homepage_header'] }
+      end
+    end
+
+    context 'when not authorised' do
+      it_behaves_like 'authorised get request', '/sign_up_form' do
+        let(:title) { data['registration_page_header'] }
+      end
+    end
+  end
+
+  context 'when POST /sign_up' do
+    before(:each) { clear_cookies }
+
+    let(:random_password) { Faker::Alphanumeric.alphanumeric(number: 10) }
+    let(:homepage_header) { data['homepage_header'] }
+    let(:not_authorised_error) { data['not_authorised_error'] }
+    let(:username_taken_error) { data['username_taken_error'] }
+    let(:registration_page_header) { data['registration_page_header'] }
+    let(:invalid_username_and_password) { data['invalid_username_or_password_error'] }
+    let(:blank_password_error) { data['blank_password_error'] }
+    let(:blank_first_name_error) { data['blank_first_name_error'] }
+    let(:blank_second_name_error) { data['blank_second_name_error'] }
+    let(:blank_username_error) { data['blank_username_error'] }
+    let(:username_length_error) { data['username_length_error'] }
+    let(:password_length_error) { data['password_length_error'] }
+    let(:blank_password_confirmation_error) { data['blank_password_confirmation_error'] }
+
+    context 'when request with valid parameters values' do
+
+      let(:successful_user_creation) { data['successful_user_creation'] }
+      let(:random_password) { Faker::Alphanumeric.alphanumeric(number: 10) }
+
+      before(:each) do
+        @username = Faker::Internet.unique.username
+        user_creds_log(@username, random_password)
+      end
+
+      # TODO implement fix shared example:
+      # it_behaves_like 'authorised POST request', '/sign_up', { username: Faker::Internet.unique.username, password: 'Test123456!', first_name: 'Test', second_name: 'User' } do
+      #   let(:success_message) { data['successful_user_creation'] }
+      # end
```

Comment:
Failed to pass user params successfully. Tried to define params in 'let', within the 'context' block and inside of the 'it_behaves_like', but with no result yet.

Reply:
To resolve this issue, you can define the parameters in a let block outside of the shared example and then use them inside the it_behaves_like block. 
let(:valid_user_params) do { necessary data }

it_behaves_like 'authorised POST request', '/sign_up', valid_user_params do .... etc.
try to do it this way

Reply:
Fixed. Needed to change the way params were defined in the shared example from
 `params` to `params = {}`